projects/angular/src/data/datagrid/API.md

```diff
@@ -0,0 +1,53 @@
+| --clr-datagrid-popover-border-color                   | Changes the border color of popup box.                                       |
+| --clr-datagrid-row-selected                           | Changes the text color of table row when user selected.                      |
+| --clr-datagrid-loading-background                     | Changes the background color of datagrid when click on toggle spinner.       |
+| --clr-datagrid-popovers-box-shadow-color              | Changes the popup box shadow color of datagrid.                              |
+| --clr-datagrid-column-switch-header-font-color        | Changes the column header text color of datagrid.                            |
+| --clr-datagrid-column-switch-header-font-hover-color  | Changes the column header text color of datagrid when hovers over element.   |
+| --clr-datagrid-detail-caret-icon-open-bg-color        | Changes the caret icon background color of datagrid.                         |
+| --clr-datagrid-detail-caret-icon-open-icon-color      | Changes the caret icon color of datagrid.                                    |
+| --clr-datagrid-placeholder-color                      | Changes the text color of datagrid placeholder.                              |
+| --clr-datagrid-placeholder-font-size                  | Changes the placeholder text size of datagrid.                               |
+| --clr-datagrid-pagination-btn-disabled-opacity        | Changes the opacity of pagination button when disabled.                      |
+| --clr-datagrid-column-toggle-border-color             | Changes the toggle button border color.                                      |
+| --clr-datagrid-column-toggle-fill-color               | Changes the background color of toggle button.                               |
+| --clr-datagrid-column-toggle-text-color               | Changes the text color of pagination toggle button.                          |
+| --clr-datagrid-column-toggle-border-hover-color       | Changes the border color of toggle button when hovers over element.          |
+| --clr-datagrid-column-toggle-fill-hover-color         | Changes the background color of toggle button when hovers over element.      |
+| --clr-datagrid-column-toggle-text-hover-color         | Changes the text color of toggle button when hovers over element.            |
+| --clr-datagrid-column-toggle-border-active-color      | Changes the border color of toggle button when active.                       |
+| --clr-datagrid-column-toggle-fill-active-color        | Changes the background color of toggle button when active.                   |
+| --clr-datagrid-column-toggle-text-active-color        | Changes the text color of pagination toggle button when active.              |
+
+
+## Class names
+
+| Class name                           | Description                                                                     |
```

Comment:
Some of the following are too obvious, like "clr-datagrid" and just identify the main datagrid element. Some other are deeply implementation specific and I see no clear way of describing them in a "reference" format. Like the ".datagrid-calculation-table".
I have no idea how they should be documented, besides the main ones like the "clr-datagrid".